esptfa_arima/Test_Management/services/analysis_doc_service.py

```
@@ -84,40 +95,60 @@ def create_analysis_document(draft: TestDraft):
+        for topic in test_topics:
+            test_mappings.append(TestTopicMapping(
                 analysis_document=document,
-                test_number=test_num,
-                defaults={'topic': topic}
-            )
+                topic=topic,
+                test_number=topic['test_number'],
+                max_score=topic['max_score'],
+            ))
+        TestTopicMapping.objects.bulk_create(test_mappings)
+        return test_mappings
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)